src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -22232,5 +22292,27 @@ public struct ValueTuple<T1, T2>
             Assert.False(tuple3.IsErrorType());
             Assert.Equal(libWithVTRef.Display, tuple3.ContainingAssembly.MetadataName.ToString());
         }
+
+        [Fact]
+        [WorkItem(17962, "https://github.com/dotnet/roslyn/issues/17962")]
+        public void TupleWithAsOperator()
+        {
+            var source = @"
+class C
+{
+    void M<T>()
+    {
+        var x = (0, null) as (int, T)?;
+        System.Console.WriteLine(x == null);
+    }
+}";
+
+            var comp = CreateCompilationWithMscorlib(source, references: new[] { ValueTupleRef, SystemRuntimeFacadeRef });
+            comp.VerifyDiagnostics(
+                // (6,17): error CS8304: The first operand of an 'as' operator may not be a tuple literal without a natural type.
```

Comment:
> I am not sure why we are not using target typing in this scenario.

The set of conversions permitted are those that don't change the representation of the runtime type of the object on the left-hand-side. See https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#the-as-operator . We *do* target type, but only the `null` value. Permitting target-typing of typeless tuples (e.g. lambdas, method groups, numeric literals, formattable string literals, tuples, etc) would contradict that intended design.